src/hashes.c

```diff
@@ -24,6 +24,10 @@ static const struct hash_info {
 	{ MH_H_BLAKE2S, "blake2s", 32 }
 };
 
+// REVIEW NOTE:
+// This could just be a table where the index into the array is the hash
+// function.  This would make the lookups much simpler.
+
```

Comment:
It can be done this way but as ANSI C doesn't support array initializes with indexes we would have to place a lot of NULLs or empty elements there.